src/lib/snark_worker/intf.ml

```diff
@@ -61,7 +61,7 @@ module type S = sig
       module V1 : sig
         type query = unit
 
-        type response = Work.Spec.t option
+        type response = Work.Spec.t option * Public_key.Compressed.t option
```

Comment:
If I understand you're description correctly, you wanted to follow up with a PR that has the worker get pushed updated when keys change -- I think we should stick with this way for the following reasons:
(1) It's simpler, so less chance of bugs
(2) The Work.Spec.t is so much larger than a public key, it's a negligible increase to the size of this payload (so there's no need to worry about any sort of space optimizations)
(3) This way we can keep the snark workers from having to register with a daemon, and they can remain just asking with this one call (so it's easy to scale up the number of workers etc)

Reply:
@bkase. Yes exactly. Okay, we can stick with this implementation then.